rust/src/state_machine/tests/impls.rs

```diff
@@ -201,4 +221,37 @@ impl MessageOwned {
             panic!("not an update message");
         }
     }
+
+    /// Extract the mask from a sum2 message
+    ///
+    /// # Panic
+    ///
+    /// Panic if this message is not a sum2 message
+    pub fn mask(&self) -> MaskObject {
+        if let Payload::Sum2(Sum2Owned { mask, .. }) = &self.payload {
+            mask.clone()
+        } else {
+            panic!("not a sum2 message");
+        }
+    }
+}
+
+impl<D> DictionaryUpdate<D> {
+    pub fn unwrap(self) -> std::sync::Arc<D> {
```

Comment:
This kind of impls makes me thing we should just use `Option`s instead of defining our own enums.

Reply:
Hm yeah on the one hand I like the idea. It would remove some code but on the other hand I think using the word `Invalidate` here make the code easier to understand.

Reply:
it's not only valuable for code understanding, but it also provides means to differentiate between "not yet available" and "already invalid", which couldn't be done with a `None`.